cloudinit/cmd/main.py

```diff
@@ -334,6 +337,90 @@ def _should_bring_up_interfaces(init, args):
     return not args.local
 
 
+def _should_wait_via_cloud_config(
+    raw_config: Optional[Union[str, bytes]]
+) -> Tuple[bool, Reason]:
+    if not raw_config:
+        return False, "no configuration found"
+
+    # If our header is anything other than #cloud-config, wait
+    possible_header: Union[bytes, str] = raw_config.strip()[:13]
+    if isinstance(possible_header, str):
+        decoded_header = possible_header
+    elif isinstance(possible_header, bytes):
+        try:
+            decoded_header = possible_header.decode("utf-8")
+        except UnicodeDecodeError:
+            return True, "Binary user data found"
+    if not decoded_header.startswith("#cloud-config"):
+        return True, "non-cloud-config user data found"
+
+    try:
+        parsed_yaml = yaml.safe_load(raw_config)
+    except Exception as e:
+        log_with_downgradable_level(
```

Comment:
I was hoping we could backport this behavior in its entirety...including the systemd ordering changes.

Reply:
Ok, I was working under the assumption that we were not going to backport the ordering changes specifically, so we'd then need to avoid the call to  systemctl start systemd-networkd-wait-online.service. If we are backporting both pkging changes and the ordering change, then no need to alter the wait on networking behavior in init-network stage. Leave as is.